src/plugins/controls/public/options_list/types.ts

```diff
@@ -22,6 +22,7 @@ export interface OptionsListComponentState {
   availableOptions?: string[];
   invalidSelections?: string[];
   validSelections?: string[];
+  existsSelectionInvalid?: boolean;
```

Comment:
Similar to the explanation for adding the  `existsSelected` property to the explicit input, I opted to store the validity of this exists selection separately from the `invalidSelections` array - this is to avoid continuous `invalidSelections.length === 1 && invalidSelections[0] === 'existsQuery'` checks

Note that, if `exists` is selected, regardless of the `exclude` property value, it is **always** the only selection - that's why the above logic works; however, it's still messier in my opinion than simply storing this value once as part of the component state so that the places that need it can easily reference this value.